### PR TITLE
trade history page

### LINKED ISSUE #320: [Mobile]: layout trade history page
use figma!

### PR DESCRIPTION
**Pull request by dreacot**
closes #320 

there's no trade history page for desktop, so i'm adding trade history page for both desktop and mobile via this PR

Screenshot from 2023-12-19 16-24-03
Screenshot from 2023-12-19 16-23-44

### CODE REVIEW

File: ui/page/exchange/order_history_page.go [hunk 1]
```diff
@@ -51,19 +62,60 @@ func NewOrderHistoryPage(l *load.Load) *OrderHistoryPage {
...
+	pg.orderDropdown = l.Theme.DropdownWithCustomPos([]cryptomaterial.DropDownItem{
+		{Text: values.String(values.StrNewest)},
+		{Text: values.String(values.StrOldest)},
+	}, values.ProposalDropdownGroup, 1, 0, true)
+
+	if pg.statusDropdown.Reversed() {
+		pg.statusDropdown.ExpandedLayoutInset.Right = values.MarginPadding10
+	} else {
+		pg.statusDropdown.ExpandedLayoutInset.Left = values.MarginPadding10
+	}
```
Comment (ukane-philemon) [inline ui/page/exchange/order_history_page.go:105]: Reversed is `true`.

```
pg.statusDropdown.ExpandedLayoutInset.Right = values.MarginPadding10

```

File: ui/page/exchange/order_history_page.go [hunk 2]
```diff
@@ -82,138 +134,252 @@ func (pg *OrderHistoryPage) OnNavigatedFrom() {
...
+	return layout.E.Layout(gtx, func(gtx C) D {
+		return layout.Flex{}.Layout(gtx,
+			layout.Rigid(pg.statusDropdown.Layout),
+			layout.Rigid(pg.orderDropdown.Layout),
+		)
 	})
 }
 
+func settingCommonDropdown(t *cryptomaterial.Theme, drodown *cryptomaterial.DropDown) {
```
Comment (ukane-philemon) [inline ui/page/exchange/order_history_page.go:374]: `useCommonDropdownSettings`?

File: ui/page/root/trade_page.go [hunk 3]
```diff
@@ -52,8 +52,13 @@ func NewTradePage(l *load.Load) *TradePage {
 			},
 		},
 	}
+	filteredTabTitles := tabTitles
+	if l.IsMobileView() {
+		// Remove the first item for mobile view
+		filteredTabTitles = filteredTabTitles[1:]
+	}
```
Comment (ukane-philemon) [inline ui/page/root/trade_page.go:59]: Please add that we are removing the DEX tab cuz it's not supported on mobile yet.

File: ui/page/root/trade_page.go [hunk 4]
```diff
@@ -120,36 +136,9 @@ func (pg *TradePage) OnNavigatedFrom() {
 // to be eventually drawn on screen.
 // Part of the load.Page interface.
 func (pg *TradePage) Layout(gtx C) D {
-	if pg.IsMobileView() {
-		return pg.layoutMobile(gtx)
-	}
 	return pg.layoutDesktop(gtx)
 }
 
 func (pg *TradePage) layoutDesktop(gtx C) D {
```
Comment (ukane-philemon) [inline ui/page/root/trade_page.go:142]: merge.

**Review by ukane-philemon — APPROVED** [review]
Nice work! Just a few nits.

- Dropdown icon is hidden for trade status dropdown on mobile view
- I think hover effect on dropdown items should cover the top and bottom space and no radius in between.
- Trade txs on mobile view look stuffed, can we reduce the text size?

**Comment by dreacot** [comment]
> I think hover effect on dropdown items should cover the top and bottom space and no radius in between.

it's seems to be that way on all dropdowns, should probably be fixed as it's own issue

**Review by ukane-philemon — APPROVED** [review]

**Pull request merged**